src/io/std_file_descriptor.cr

```diff
@@ -0,0 +1,26 @@
+require "./syscall"
+require "./file_descriptor"
+
+class IO::StdFileDescriptor < IO::FileDescriptor
+  def initialize(@fd : Int32)
+    @closed = false
+
+    # If we have a TTY for stdin/out/err, it is a shared terminal.
+    # We need to reopen it to use O_NONBLOCK without causing other programs to break
+    if tty?
+      # Figure out the terminal TTY name
+      path = UInt8[256]
```

Comment:
```crystal
path = uninitialized UInt8[256]
```

What you wrote creates a slice of size 1 with the value 256. You are lucky it doesn't somehow segfaults.